### PR TITLE
[CXF-8162] JWE with multiple recipients does not work for AES CBC Encryption

### PR DESCRIPTION
**Pull request by frelibert**
Fixed:
- System.arraycopy src position
- exposed AesCbcContentEncryptionAlgorithm to make cek and iv reusable for all recipients

### CODE REVIEW

File: rt/rs/security/jose-parent/jose/src/main/java/org/apache/cxf/rs/security/jose/jwe/AesCbcHmacJweEncryption.java [hunk 1]
```diff
@@ -19,66 +19,50 @@
...
     protected byte[] getActualCek(byte[] theCek, String algoJwt) {
         return doGetActualCek(theCek, algoJwt);
     }
+    
     protected static byte[] doGetActualCek(byte[] theCek, String algoJwt) {
-        int size = getFullCekKeySize(algoJwt) / 2;
+        int size = AesCbcContentEncryptionAlgorithm.getFullCekKeySize(algoJwt) / 2;
         byte[] actualCek = new byte[size];
-        System.arraycopy(theCek, size, actualCek, 0, size);
+        System.arraycopy(theCek, 0, actualCek, 0, size);
```
Comment (coheigea) [inline rt/rs/security/jose-parent/jose/src/main/java/org/apache/cxf/rs/security/jose/jwe/AesCbcHmacJweEncryption.java]: I think this change may not be correct. The spec says that for AES_128_CBC_HMAC_SHA_256 the input key must be 32 bytes, and the test in JweJsonProducerTest.testMultipleRecipientsA128CBCHS256GivenCek is passing in a 16 byte key hence the error:

"The input key K is 32 octets long."
https://tools.ietf.org/html/rfc7518#section-5.2.3

**Comment by frelibert** [comment]
Thank you for the feedback.
You were right. The problem with the array indexoutofbound was due to an invalid inputKey.
I reverted that change and I've added a check for proper keysize to have better exceptionhandling.

File: rt/rs/security/jose-parent/jose/src/main/java/org/apache/cxf/rs/security/jose/jwe/AesCbcContentEncryptionAlgorithm.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+import javax.crypto.spec.IvParameterSpec;
+
+import org.apache.cxf.rs.security.jose.jwa.AlgorithmUtils;
+import org.apache.cxf.rs.security.jose.jwa.ContentAlgorithm;
+
+public class AesCbcContentEncryptionAlgorithm extends AbstractContentEncryptionAlgorithm {
+    
+    static final Map<String, String> AES_HMAC_MAP;
+    static final Map<String, Integer> AES_CEK_SIZE_MAP;
```
Comment (coheigea) [inline rt/rs/security/jose-parent/jose/src/main/java/org/apache/cxf/rs/security/jose/jwe/AesCbcContentEncryptionAlgorithm.java]: I'd prefer to make these private and add getter methods. Either that or make them unmodifiable maps.

**Comment by frelibert** [comment]
Ok, thanks.
I had committed another fix (for the private property issue) but I see you fixed that yourself by now.
I'll remove my forked branch

**Comment by coheigea** [comment]
@frelibert thanks for the PR!

**Pull request merged**